ext/cl_khr_fp16.asciidoc

```diff
@@ -1788,3 +1788,126 @@ precision floating-point values and performing the operation in single
 precision floating-point.
 In this case, the implementation will use the_ `half` _scalar or vector data
 type as a storage only format_.
+
+[[cl_khr_fp16-additions-to-chapter-8-of-the-opencl-2.0-specification]]
+=== Additions to Chapter 8 of the OpenCL 2.0 C Specification
+
+Add new sub-sections to _section 8.3.1. Conversion rules for normalized integer channel data types_:
+
+[[cl_khr_fp16-converting-normalized-integer-channel-data-types-to-floating-point-values]]
+==== Converting normalized integer channel data types to half precision floating-point values
+
+For images created with image channel data type of `CL_UNORM_INT8` and
+`CL_UNORM_INT16`, *read_imagef* will convert the channel values from an
+8-bit or 16-bit unsigned integer to normalized half precision
+floating-point values in the range [`0.0h`, `1.0h`].
+
+For images created with image channel data type of `CL_SNORM_INT8` and
+`CL_SNORM_INT16`, *read_imagef* will convert the channel values from an
+8-bit or 16-bit signed integer to normalized half precision floating-point
+values in the range [`-1.0h`, `1.0h`].
+
+These conversions are performed as follows:
+
+`CL_UNORM_INT8` (8-bit unsigned integer) {rightarrow} `half`
+
+  :: normalized `half` value = `round_to_half(c / 255)`
+
+`CL_UNORM_INT_101010` (10-bit unsigned integer) {rightarrow} `half`
+
+  :: normalized `half` value = `round_to_half(c / 1023)`
```

Comment:
This is equivalent to the SPIR-V environment spec, but there the formulas use a mathematical formatting whereas here it's more C like.

Would `round_to_half((half)c / 1023.0h)` be clearer?   I don't actually know if that is actually correct though (compared to thinking of the divide as infinite-precision).

Reply:
This is the only change I haven't made, primarily because I think we should clarify that this is intended to be an infinite-precision division for all conversions, both to `half` and to `float` in the OpenCL C spec and SPIR-V Environment spec.  I'd like to make these changes as part of the resolution for #97 and #183.  Does this sound reasonable?

Reply:
I didn't even write what my original issue was: read with C syntax meant I read it as an integer divide that will always be `0` or `1`.

But completely reasonable to wait until these cases are all resolved together.